pkg/task/task.go

```diff
@@ -77,3 +91,40 @@ func (t *Task) State() DatedState {
 	}
 	return t.States[len(t.States)-1]
 }
+
+func (t *Task) ParseCreatedBy() string {
+	res := strings.Split(t.CreatedBy, ";")
+
+	if len(res) == 2 {
+		url := res[1]
+		title := res[0]
+		return fmt.Sprintf(`<a href="%s" target="_blank">%s</a>`, url, title)
+	}
+
+	return t.CreatedBy
+}
+
+// CreatedByCommit parses the task.CreatedBy field in case it contains a ;
+// https://github.com/filecoin-project/oni/commit/d8b5a13f1fc758c674ffcee6dbd7f13540997283
+// Currently CircleCI utilises the following format for the `user` field in the `.env.toml`
+// user="filecoin-project/oni -> '$CIRCLE_BRANCH';https://github.com/filecoin-project/oni/commit/'$CIRCLE_SHA1'
```

Comment:
Can't we break this down into different fields? The `RunEnv` from prototype versions of Testground had fields for branch, commit, etc. It doesn't need to be part of the `RunEnv`, but I think modelling those source control attributes in Testground is the way forward.

Reply:
Yes, that's an absolute hack right now. I wasn't even sure that we want this metadata about run to be propagated as part of `user`, but didn't want to add more flags, so piggy-backed on the `user` field.

Reply:
This is about propagating the `commit`, `branch`, etc. from the `testground client` to the `testground daemon` however, I don't see how `RunEnv` applies here.

Reply:
Refactoring this, so that we can include `metadata-commit`, `metadata-branch` and `metadata-repo` from the CLI when submitting a composition. These would be used by CircleCI for example.